src/views/admin/CreateExamRegForm.vue

```diff
@@ -0,0 +1,144 @@
+            <span class="font-bold text-sm ml-1">Back</span>
+        </button>
+        <h1 class="text-2xl font-medium text-gray-700 mb-8">{{ heading }}</h1>
+        <div>
+            <ExRegInfoSection v-model:name="examRegName" v-model:valid="nameValid" :existing-names="existingExamRegNames" />
+            <ExRegModuleSection v-model:modules="selectedModules" :existing-modules="existingModules" />
+            <section class="border-t-2 py-8 border-gray-400 lg:mt-8">
+                <div class="hidden sm:flex justify-between">
+                    <div class="flex justify-end items-center">
+                        <button id="cancel" type="button" class="w-32 mr-6 btn btn-blue-secondary" @click="back">Cancel</button>
+                        <button
+                            id="createExamReg"
+                            :disabled="!canCreate"
+                            style="width: 18rem"
+                            class="btn btn-blue-primary"
+                            @click="createExamReg"
+                        >
+                            Create Examination Regulation
+                        </button>
+                    </div>
+                </div>
+
+                <!-- different button layout for mobile -->
+                <div class="sm:hidden">
+                    <button id="mobileCancel" type="button" class="mb-4 w-full btn btn-blue-secondary" @click="back">Cancel</button>
+                    <button id="mobileCreateCourse" :disabled="!canCreate" class="mb-4 w-full btn btn-blue-primary" @click="createExamReg">
+                        Create Examination Regulation
+                    </button>
+                </div>
+            </section>
+            <unsaved-changes-modal ref="unsavedChangesModal" />
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import { ref, computed, onBeforeMount } from "vue";
+    import UnsavedChangesModal from "@/components/modals/UnsavedChangesModal.vue";
+    import Router from "@/use/router";
+    import Module from "@/api/api_models/exam_reg_management/Module";
+    import ExaminationRegulation from "@/api/api_models/exam_reg_management/ExaminationRegulation";
+    import ExRegInfoSection from "@/components/exreg/ExRegInfoSection.vue";
+    import ExRegModuleSection from "@/components/exreg/ExRegModuleSection.vue";
+    import LoadingComponent from "@/components/common/loading/Spinner.vue";
+    import ExaminationRegulationManagement from "@/api/ExaminationRegulationManagement";
+    import GenericResponseHandler from "@/use/helpers/GenericResponseHandler";
+    import ValidationResponseHandler from "@/use/helpers/ValidationResponseHandler";
+    import { useToast } from "@/toast";
+
+    export default {
+        name: "CreateExamRegForm",
+        components: {
+            ExRegModuleSection,
+            UnsavedChangesModal,
+            ExRegInfoSection,
+            LoadingComponent,
+        },
+        setup: function () {
```

Comment:
```suggestion
        setup() {
```
Just a feeling of keeping the files consistent :)